aptos-indexer-processors-sdk/sdk/src/common_steps/order_by_version_step.rs

```diff
@@ -0,0 +1,185 @@
+use crate::{
+    traits::{
+        pollable_async_step::PollableAsyncRunType, NamedStep, PollableAsyncStep, Processable,
+    },
+    types::transaction_context::TransactionContext,
+    utils::errors::ProcessorError,
+};
+use ahash::AHashMap;
+use anyhow::Result;
+use async_trait::async_trait;
+use std::time::Duration;
+
+pub struct OrderByVersionStep<Input>
+where
+    Self: Sized + Send + 'static,
+    Input: Send + 'static,
+{
+    pub ordered_versions: Vec<TransactionContext<Input>>,
```

Comment:
nit: in case we have increasing transaction volume, to avoid bottlenecks, should we consider optimizing the storage of unordered version like using a concurrent data structure?

Reply:
Is there a concurrent data structure you had in mind? If it's run concurrently, it will need to acquire a lock around around the unordered and ordered versions. With increasing transaction volume, we can also make it return output more frequently.

Reply:
one thing that comes into my mind is DashMap. it should allow concurrent reads and writes without having to manually lock the entire data structure. https://docs.rs/dashmap/latest/dashmap/. but nothing to urgent right now! we can revisit if it really becomes the bottlenecks